src/System.Private.CoreLib/src/System/Exception.CoreCLR.cs

```diff
@@ -324,7 +316,7 @@ internal void RestoreDispatchState(in DispatchState dispatchState)
                     //
                     // Since EDI can be created at various points during exception dispatch (e.g. at various frames on the stack) for the same exception instance,
                     // they can have different data to be restored. Thus, to ensure atomicity of restoration from each EDI, perform the restore under a lock.
-                    lock (s_DispatchStateLock)
+                    lock (_stackTraceLock)
```

Comment:
Maybe fallback to global lock here as well?

Reply:
Added new global lock with fallback via property